### ISSUE TITLE
🐛 GitHub Contributions Calendar is dead 🪦

### ISSUE THREAD

**Issue by BrookJeynes** [SSWConsulting/SSW.People#433]
@bradystroud @JackDevAU @adamcogan 

To Myself,

The Github Contributions Calendar is not working.

As per my conversation with @bradystroud and @JackDevAU, we believe this is an issues with Heroku removing its free tier. 

- [x] Fix to the issue

<img width="1624" alt="Screen Shot 2022-09-15 at 12 24 32" src="https://user-images.githubusercontent.com/25432120/190298615-6b71e260-9839-4473-ad53-1b801e91a3ca.png">

**Figure: Error :(**

**Comment by BrookJeynes** [comment]
Update as of 16th September:

The issue seems to be arising due to the `react-github-calendar` package being on an older version, version `^2.0.2`. This version used a Heroku server for something and has since been closed with the new Heroku T&Cs.

I've attempted to update the package to version `3.3.3` (the latest at the current time) however, this seems to rely on `react-activity-calendar` which uses Javascript syntax not supported in running Node version (more specifically optional chaining syntax).

By adding the below snippet to the `package.json` file, optional chaining syntax seemed to be supported and the issue was fixed when the project was run via `yarn develop`.
```json
"resolutions": {
    "acorn": "8.0.1"
  }
``` 

This however, does not seem to be the complete fix as when running `yarn build` (as noticed when the PR failed) this new error appears:

```bash
  267 |     const date = format__default["default"](parseISO__default["default"](contribution.date), dateFormat);
  268 |     const tooltip = (_labels$tooltip = labels.tooltip) !== null && _labels$tooltip !== void 0 ? _labels$tooltip : DEFAULT_LABELS.tooltip;
> 269 |     return tooltip.replaceAll('{{count}}', String(contribution.count)).replaceAll('{{date}}', date);
      | ^
  270 |   }
  271 |
  272 |   function getEventHandlers(data) {


  WebpackError: TypeError: tooltip.replaceAll is not a function
```

**Comment by BrookJeynes** [comment]
To fix the Github contributions calendar, we need to upgrade the package. 

The main issue with this is that the newer package version uses unsupported Javascript syntax in the current Node version. To fix this, we are trying to upgrade node to a higher version, this is unfortunately difficult due to how outdated some of the packages are. A majority of the packages will need to be updated to support the newer Javascript syntax and Node version.

More specifically, the issue below turns out to still be coming from the `react-activity-calendar` module and is occurring due to the `.replaceAll()` method not being implemented in Node 14.

```bash
  267 |     const date = format__default["default"](parseISO__default["default"](contribution.date), dateFormat);
  268 |     const tooltip = (_labels$tooltip = labels.tooltip) !== null && _labels$tooltip !== void 0 ? _labels$tooltip : DEFAULT_LABELS.tooltip;
> 269 |     return tooltip.replaceAll('{{count}}', String(contribution.count)).replaceAll('{{date}}', date);
      | ^
  270 |   }
  271 |
  272 |   function getEventHandlers(data) {


  WebpackError: TypeError: tooltip.replaceAll is not a function
```

The next step we're currently trying is upgrading Node from v14 to v16 and all related packages.

**Comment by DevStarOps** [comment]
@BrookJeynes 

Should we not hide that section until we have a fix?

**Comment by bradystroud** [comment]
## Update:

**TLDR:** I have made good progress, but there is at least a day of work remaining

> 1. As per @DevStarOps suggestion, temporarily hide the section so the error text is not visible

✅ Done - #438

> 2. Update all packages in the project, fixing breaking changes as I go

🚧 In progress - after updating the project, there were 5 major issues

**✅ I have fixed 3**
1. Body of profile content 
   - Profiles with no skill images work
2. Breadcrumb not working - vertical list instead of horizontal
   - Caused by upgrading `gatsby-plugin-breadcrumb`
   - Adding some CSS fixed it
3. Youtube carousel - new version no longer supports multiple slides per page
   - Locked at the previous version


**❌ There are 2 remaining**

4. MegaMenu is not compatible
    - Probably just need to upgrade React version on the megamenu
5. Index page images not showing up
    - This was mostly fixed following this migration guide https://www.gatsbyjs.com/docs/reference/release-notes/image-migration-guide/
    - There is still a bug where images are cut off when fullscreen
    - These docs might help https://www.gatsbyjs.com/docs/reference/release-notes/v3.0/#gatsby-plugin-image100
![Screenshot of ](https://user-images.githubusercontent.com/38869720/201432079-91b5301e-5aeb-4f23-a02c-06a4061c1610.png)
**Figure: Profile image not responsive**

> 3. The new version of the GitHub contributions widget should work and the issue will be resolved

Done - As expected, the new version of the package fixes the issue.

I estimate the remaining issues can be resolved with another day of work

**Comment by bradystroud** [comment]
## Update

> 5. Index page images not showing up

✅ Done - This was fixed by adusting the graphql query in this commit https://github.com/SSWConsulting/SSW.People/commit/b2958679a36c5eeae26ba563dd11045c3884777a

**Comment by BrookJeynes** [comment]
## Update

> 4. MegaMenu is not compatible

✅ Done - The MegaMenu is now compatible with the People
❌ New issue -  The MegaMenu fails to fetch the data contained within itself on build.

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/25432120/203931495-bdddaaff-73a8-4fec-bc81-eb9fad6eaae5.png">
Figure: MegaMenu in People but failing to fetch data

**Comment by wicksipedia** [comment]
Wasn't able to repro the broken image bug but I did notice that the hover effect isn't working on staging.

I made a little video to demo, hope this helps.


https://user-images.githubusercontent.com/600044/207172040-5f79a306-81a1-496c-ad6b-911fb56e61ad.mp4

**Comment by BrookJeynes** [comment]
## Update

> The user profiles on the main peoples page are not loading

✅ Done - The profile pictures are now loading successfully. This was fixed by following the solution found here https://stackoverflow.com/questions/46044933/iis-server-blocking-access-to-webp-files

❌ There is now 1 remaining issue
1. The hover effect on the profiles isn't working as mentioned here https://github.com/SSWConsulting/SSW.People/issues/433#issuecomment-1347452281.

I will continue to investigate this issue

**Comment by BrookJeynes** [comment]
## Update

> The hover effect on the profiles isn't working as mentioned here https://github.com/SSWConsulting/SSW.People/issues/433#issuecomment-1347452281.

✅ Done - The hover effect is now fixed by manually overriding Gatsby styling.

> Fix to the issue

✅ Done - All issues have been resolved and pushed to prod.

**Comment by AlexRothwell** [comment]
It looks like something might be wrong with the sorting:
![image](https://user-images.githubusercontent.com/7465195/207482946-bfb8b044-0df4-47d9-8b2c-9263b26a7110.png)

On initial load, the profiles are displayed alphabetically then sorted but the images aren't updated to reflect this. It's fine if you navigate to a profile and then hit back

**Comment by BrookJeynes** [comment]
Hey @AlexRothwell,

This is already a know issue - https://github.com/SSWConsulting/SSW.People/issues/378
Please add your comment to the above issue.
